### PR TITLE
python@3.13 modify sysconfig

### PR DESCRIPTION
**Pull request by johng**
Retains abi flags need for no-gil guild

### CODE REVIEW

**Comment by johng** [comment]
Is applied here https://github.com/Homebrew/homebrew-core/pull/193855/files#diff-ba8fa367dd5d457c53d4e8e60e20e20e48753c59339e428132229b0fbdd7bef2

**Comment by johng** [comment]
Compatible with both gil and non gil build

File: python/3.13-sysconfig.diff [hunk 1]
```diff
@@ -17,7 +17,7 @@ index 80aef34..4af9376 100644
 +        'data': '@@HOMEBREW_PREFIX@@',
 +        },
      }
-
+
```
Comment (carlocab) [inline python/3.13-sysconfig.diff:20]: Do we need this whitespace change?

File: python/3.13-sysconfig.diff [hunk 2]
```diff
@@ -28,15 +28,15 @@ index 80aef34..4af9376 100644
              'home': 'posix_home',
              'user': 'osx_framework_user',
          }
-
+
```
Comment (carlocab) [inline python/3.13-sysconfig.diff:31]: And this one?

File: python/3.13-sysconfig.diff [hunk 3]
```diff
@@ -28,15 +28,15 @@ index 80aef34..4af9376 100644
...
      return {
 -        'prefix': 'posix_prefix',
 +        'prefix': 'osx_framework_library',
          'home': 'posix_home',
          'user': 'posix_user',
      }
-
-
+ 
+
```
Comment (carlocab) [inline python/3.13-sysconfig.diff:39]: Ditto.

File: python/3.13-sysconfig.diff (hunk 1 above)
Comment (johng) [reply python/3.13-sysconfig.diff:20]: No trying to remove, have been doing `git diff --ignore-blank-lines --ignore-all-space --ignore-cr-at-eol` piping directly to the patch file here.

File: python/3.13-sysconfig.diff [hunk 4]
```diff
@@ -7,8 +7,8 @@ index 80aef34..4af9376 100644
          'data': '{base}',
          },
 +    'osx_framework_library': {
-+        'stdlib': '{installed_base}/{platlibdir}/python{py_version_short}',
-+        'platstdlib': '{platbase}/{platlibdir}/python{py_version_short}',
++        'stdlib': '{installed_base}/{platlibdir}/python{py_version_short}{abi_thread}',
++        'platstdlib': '{platbase}/{platlibdir}/python{py_version_short}{abi_thread}',
 +        'purelib': '@@HOMEBREW_PREFIX@@/lib/python{py_version_short}/site-packages',
 +        'platlib': '@@HOMEBREW_PREFIX@@/{platlibdir}/python{py_version_short}/site-packages',
```
Comment (Bo98) [inline python/3.13-sysconfig.diff:13]:
```suggestion
+        'purelib': '@@HOMEBREW_PREFIX@@/lib/python{py_version_short}{abi_thread}/site-packages',
+        'platlib': '@@HOMEBREW_PREFIX@@/{platlibdir}/python{py_version_short}{abi_thread}/site-packages',
```

**Review by Bo98 — APPROVED** [review]

**Pull request merged**